test/appRelationsDiscovery_test.js

```diff
@@ -0,0 +1,64 @@
+const flightQueuePubData = '{"Flight Monitor Service":{"_json":{"description":"Queues a flight in order to retrieve status\\n","publish":{"summary":"Subscribe about the status of a given flight","message":{"summary":"Requets to queue a flight to be monitored","payload":{"type":"object","properties":{"flight":{"type":"object","properties":{"carrierCode":{"type":"string","description":"2 to 3-character IATA carrier code","example":"LH","x-parser-schema-id":"<anonymous-schema-17>"},"flightNumber":{"type":"integer","minimum":1,"description":"1 to 4-digit number of the flight","example":"193","x-parser-schema-id":"<anonymous-schema-18>"},"scheduledDepartureDate":{"type":"string","format":"date-time","description":"scheduled departure date of the flight, local to the departure airport.","example":"2020-10-20","x-parser-schema-id":"<anonymous-schema-19>"}},"x-parser-schema-id":"<anonymous-schema-16>"},"user":{"type":"object","properties":{"userName":{"type":"string","minimum":1,"description":"user name","example":"John Smith","x-parser-schema-id":"<anonymous-schema-21>"},"phoneNumber":{"type":"string","minimum":5,"description":"phone number where notifications will be received.","example":"+13451235","x-parser-schema-id":"<anonymous-schema-22>"}},"x-parser-schema-id":"<anonymous-schema-20>"}},"x-parser-schema-id":"<anonymous-schema-15>"},"x-parser-original-schema-format":"application/vnd.aai.asyncapi;version=2.0.0","x-parser-original-payload":{"type":"object","properties":{"flight":{"type":"object","properties":{"carrierCode":{"type":"string","description":"2 to 3-character IATA carrier code","example":"LH"},"flightNumber":{"type":"integer","minimum":1,"description":"1 to 4-digit number of the flight","example":"193"},"scheduledDepartureDate":{"type":"string","format":"date-time","description":"scheduled departure date of the flight, local to the departure airport.","example":"2020-10-20"}}},"user":{"type":"object","properties":{"userName":{"type":"string","minimum":1,"description":"user name","example":"John Smith"},"phoneNumber":{"type":"string","minimum":5,"description":"phone number where notifications will be received.","example":"+13451235"}}}}},"schemaFormat":"application/vnd.aai.asyncapi;version=2.0.0","x-parser-message-parsed":true,"x-parser-message-name":"flightQueue"}}}}}';
+let flightServiceDocs;
+let output;
+let obj;
+(async() => {
+  flightServiceDocs = getAsyncApiExamples();
+  output = await getRelations(flightServiceDocs);
+  obj = mapToObject(output);
+})();
+
+describe('appRelationDiscovery', function() {
+  it('should return correct slug', function() {
+    expect(JSON.stringify(Object.keys(obj))).to.deep.equal('["mqtt://localhost:1883,mqtt"]');
```

Comment:
is it needed to convert map to plain object and then stringify to do the assertion?
I think it would be easier to operate directly on map output, for example:
```
expect(output.get('mqtt://localhost:1883,mqtt')).to.not.be.undefined;
```

have a look at 2 possible ways to do these tests without any transformation of maps to object and string
```
describe('appRelationDiscovery', function() {
  let correctSlug;
  let slugOutput;
  let output;

  before(async function() {
    const flightServiceDocs = getAsyncApiExamples();
    output = await getRelations(flightServiceDocs);
    correctSlug = 'mqtt://localhost:1883,mqtt';
    slugOutput = output.get(correctSlug);
  });

  it('should return correct slug', function() {
    const outputIterator = output.keys();

    expect(slugOutput).to.not.be.undefined;
    //or
    expect(outputIterator.next().value).to.equal(correctSlug);
  });

  it('should return correct channels', function() {
    const correctChannelUpdate = 'flight/update';
    const correctChannelQueue = 'flight/queue';
    const channelsIterator = slugOutput.keys();

    expect(slugOutput.get(correctChannelUpdate)).to.not.be.undefined;
    expect(slugOutput.get(correctChannelQueue)).to.not.be.undefined;
    //or
    expect(channelsIterator.next().value).to.equal(correctChannelUpdate);
    expect(channelsIterator.next().value).to.equal(correctChannelQueue);
  });
});
```

what do you think

Reply:
never thought like this way tbh, and I think this would be the more correct approach :)